# Assignment for Week 1    


# New Section

## Purpose:      
This assignment is intended to help you do the following:       
1.	Open a GitHub account for your own use and for your career advancement (if you already do not have one).     
2.	Submit information and code in GitHub following best practices from the Udacity course.     
3.	Explore a dataset thoroughly.      
4.	Find and print descriptive information from the dataset.        


## Action Items:           
Open a GitHub account to submit your work. If you are new to GitHub, follow the videos and instructions on the following link to understand how to post things on GitHub and start your own account. https://blog.udacity.com/2015/06/a-beginners-git-github-tutorial.html and/or https://guides.github.com/activities/hello-world/     

1.   Select the dataset of your choice from the list 
provided <a href="https://github.com/frankData612">here.</a>   
2.   Load the dataset using Jupyter notebook and explore the data by using the following.


-	Show the .head() and .tail() of the data set. Specify the number of rows to print using .head() and .tail().      
-	Print the column names.      
-	Print to see what is the type of your data set    
-	Check number of rows and columns.
-	Use groupby() and find the mean on your selected colum/s.
-	Add a README file in GitHub where you explain everything you did.


In [0]:
# Import the required Dataset from Franklin github site
import pandas as pd
df = pd.read_csv('https://raw.githubusercontent.com/frankData612/data_612/master/State_Drug_Utilization_Data_2010/State_Drug_Utilization_Data_2010.csv')

In [0]:
# Show the first 10 Rows from the 2010 State Drug Utilization Dataset
df.head(10)


Unnamed: 0 Utilization Type  ...             Location          NDC
0      128037             MCOU  ...  (31.1801, -91.8749)  37205013705
1      491755             FFSU  ...  (39.3498, -75.5148)  62756076388
2     2568076             FFSU  ...  (44.2853, -99.4632)  46987032311
3      491263             FFSU  ...  (43.3504, -84.5603)    781518501
4      836489             FFSU  ...                  NaN  51552080906
5     2468555             FFSU  ...   (37.768, -78.2057)    143178501
6       73349             FFSU  ...   (37.768, -78.2057)  45802025742
7     1166159             FFSU  ...  (44.6074, -69.3977)    527058610
8     2326672             FFSU  ...  (35.6411, -79.8431)  68382003001
9      413825             FFSU  ...  (44.6074, -69.3977)    172436660

[10 rows x 21 columns]

In [0]:
# Show the last 10 Rows of data from the 2010 State Drug Utilization Dataset
df.tail(10)

Unnamed: 0 Utilization Type  ...             Location          NDC
156210     2766345             FFSU  ...  (32.9866, -83.6487)     93081101
156211      725194             FFSU  ...   (36.17, -119.7462)  67253090011
156212     2143770             FFSU  ...   (42.0046, -93.214)  68382034201
156213      785538             FFSU  ...  (35.6411, -79.8431)  49348029598
156214      317999             FFSU  ...  (41.1289, -98.2883)  50111064702
156215     2752214             MCOU  ...                  NaN  63868001701
156216     2519195             MCOU  ...  (44.6074, -69.3977)    121048910
156217      815768             MCOU  ...  (42.2373, -71.5314)  60432013325
156218     2029158             FFSU  ...  (38.8964, -77.0262)     88110947
156219     2799238             FFSU  ...  (38.8964, -77.0262)     29152725

[10 rows x 21 columns]

In [0]:
# Print the Column Names from the 2010 State Drug Utilization Dataset
print(df.columns)

Index(['Unnamed: 0', 'Utilization Type', 'State', 'Labeler Code',
       'Product Code', 'Package Size', 'Year', 'Quarter', 'Product Name',
       'Units Reimbursed', 'Suppression Used', 'Number of Prescriptions',
       'Total Amount Reimbursed', 'Medicaid Amount Reimbursed',
       'Non Medicaid Amount Reimbursed', 'Quarter Begin', 'Quarter Begin Date',
       'Latitude', 'Longitude', 'Location', 'NDC'],
      dtype='object')


In [0]:
# Print the data types of the 2010 State Drug Utilization Dataset
df.dtypes

Unnamed: 0                          int64
Utilization Type                   object
State                              object
Labeler Code                        int64
Product Code                        int64
Package Size                        int64
Year                                int64
Quarter                             int64
Product Name                       object
Units Reimbursed                  float64
Suppression Used                     bool
Number of Prescriptions             int64
Total Amount Reimbursed           float64
Medicaid Amount Reimbursed        float64
Non Medicaid Amount Reimbursed    float64
Quarter Begin                      object
Quarter Begin Date                 object
Latitude                          float64
Longitude                         float64
Location                           object
NDC                                 int64
dtype: object

In [0]:
# Let's see what information the df.info() syntax gives
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 156220 entries, 0 to 156219
Data columns (total 21 columns):
 #   Column                          Non-Null Count   Dtype  
---  ------                          --------------   -----  
 0   Unnamed: 0                      156220 non-null  int64  
 1   Utilization Type                156220 non-null  object 
 2   State                           156220 non-null  object 
 3   Labeler Code                    156220 non-null  int64  
 4   Product Code                    156220 non-null  int64  
 5   Package Size                    156220 non-null  int64  
 6   Year                            156220 non-null  int64  
 7   Quarter                         156220 non-null  int64  
 8   Product Name                    156166 non-null  object 
 9   Units Reimbursed                156220 non-null  float64
 10  Suppression Used                156220 non-null  bool   
 11  Number of Prescriptions         156220 non-null  int64  
 12  Total Amount Rei

In [0]:
# Print/Check the number of Rows and Columns in the 2010 State Drug Utilization Dataset 
df.shape

(156220, 21)

From the syntax above you can see that we have 156,220 rows of data by 21 columns and no null values.

In [0]:
# Let see some Stats from our 2010 State Drug Utilization Dataset
import sys
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import pandas.util.testing as tm
from pandas import set_option
plt.style.use('ggplot')
set_option('display.width', 100)
set_option('precision', 2)

print("SUMMARY STATISTICS OF NUMERIC COLUMNS")
print()
print(df.describe().T)


SUMMARY STATISTICS OF NUMERIC COLUMNS

                                   count      mean  ...       75%       max
Unnamed: 0                      156220.0  1.56e+06  ...  2.34e+06  3.12e+06
Labeler Code                    156220.0  2.50e+04  ...  5.51e+04  9.92e+04
Product Code                    156220.0  1.80e+03  ...  2.65e+03  9.99e+03
Package Size                    156220.0  2.21e+01  ...  3.20e+01  9.90e+01
Year                            156220.0  2.01e+03  ...  2.01e+03  2.01e+03
Quarter                         156220.0  2.57e+00  ...  4.00e+00  4.00e+00
Units Reimbursed                156220.0  2.08e+04  ...  3.37e+03  6.01e+07
Number of Prescriptions         156220.0  3.17e+02  ...  7.00e+01  3.89e+05
Total Amount Reimbursed         156220.0  2.19e+04  ...  2.15e+03  5.03e+07
Medicaid Amount Reimbursed      156220.0  2.09e+04  ...  2.05e+03  4.88e+07
Non Medicaid Amount Reimbursed  156220.0  9.92e+02  ...  1.30e+01  1.44e+07
Latitude                        145416.0  3.74e+0

/usr/local/lib/python3.6/dist-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
  import pandas.util.testing as tm


In [0]:
# Now lets use the "groupby" function to see specific column "mean" values
df.groupby(['Year', 'Quarter']) [['Number of Prescriptions', 'Medicaid Amount Reimbursed', 'Non Medicaid Amount Reimbursed']].mean()

Number of Prescriptions  ...  Non Medicaid Amount Reimbursed
Year Quarter                           ...                                
2010 1                         270.91  ...                          966.43
     2                         335.03  ...                         1000.98
     3                         316.29  ...                          707.30
     4                         338.12  ...                         1284.96

[4 rows x 3 columns]

**Homework Assignment Number 1/Data612 Overiew and Item Discussion: ****
We start by opening Jupyter Notebook and loading the 2010 State Drug Utilization dataset from the https://raw.githubusercontent.com/frankData612/master/State_Drug_Utilization_Data_2010/State_Drug_Utilization_Data_2010.csv file (Item 1).

Next load the dataset into pandas and import necessary pandas functions to utilize the dataset (Item 2).

Next, we show the top 10 rows and the bottom 10 rows using the **df.head(10)** and **df.tail(10)** syntax (Item 3).

We follow this by printing the Column Names using the **print(df.columns)** syntax (Item 4).

Next, we print the data types of the 2010 State Drug Utilization dataset using the **df.dtypes** **and df.info()** syntax (Item 5).

This is followed by printing the shape of our 2010 State Drug Utilization datset using the **df.shape** syntax.

Next, we printout the Summary Statistics using a series of pandas syntax commands (above).

Finally, we use the **df.groupby([' ', ' ']).mean()** syntax to review specific column "mean" values grouped by year and quarter.